# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'до 30 тыс'
        elif 30001 <= income <= 50000:
            return 'до 50 тыс'
        elif 50001 <= income <= 200000:
            return 'до 200 тыс'
        elif 200001 <= income <= 1000000:
            return 'до миллиона'
        elif income >= 1000001:
            return 'более миллиона'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)
data

children  days_employed  dob_years education  education_id  \
0             1    8437.673028         42    высшее             0   
1             1    4024.803754         36   среднее             1   
2             0    5623.422610         33   среднее             1   
3             3    4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21520         1    4529.316663         43   среднее             1   
21521         0  343937.404131         67   среднее             1   
21522         1    2113.346888         38   среднее             1   
21523         3    3112.481705         38   среднее             1   
21524         2    1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

       total_income                       purpose total_income_category  
0            253875                 покупка жилья           до миллиона  
1            112080       приобретение автомобиля            до 200 тыс  
2            145885                 покупка жилья            до 200 тыс  
3            267628    дополнительное образование           до миллиона  
4            158616               сыграть свадьбу            до 200 тыс  
...             ...                           ...                   ...  
21520        224791             операции с жильем           до миллиона  
21521        155999          сделка с автомобилем            до 200 тыс  
21522         89672                  недвижимость            до 200 тыс  
21523        244093  на покупку своего автомобиля           до миллиона  
21524         82047         на покупку автомобиля            до 200 тыс  

[21331 rows x 13 columns]

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
data_group = data.groupby('children', as_index = False).agg({'debt':['count', 'sum', 'mean']})

In [23]:
data_group.columns = ['Количество детей', 'Число заёмщиков', 'Число должников', 'Доля должников']

In [24]:
data_sort = data_group.sort_values(by='Доля должников')
data_sort

Количество детей  Число заёмщиков  Число должников  Доля должников
5                 5                9                0        0.000000
0                 0            14091             1063        0.075438
3                 3              330               27        0.081818
1                 1             4808              444        0.092346
2                 2             2052              194        0.094542
4                 4               41                4        0.097561

<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkMagenta'>👉 Можно сделать вот так, через функцию.</font>

In [25]:
# Комментарий ревьюера
# Создаем копию датасета, что бы не работать с исходным
temp = data.copy()
# Напишем функцию, так как все задачи аналогичны
def que(category):
    data_temp = temp.pivot_table(index=category, values='debt', aggfunc=['count', 'sum', 'mean'])
    data_temp.columns = ['Всего кредитополучателей', 'Всего должников', 'Доля должников']
    # Оформим таблицу цветным градиентом, но можно ее вывести и просто display(data_temp)
    display(data_temp.style.format("{:.3f}").background_gradient(cmap='Blues', axis=0))
    
que('children')

**Вывод:** если отбросить малочисленные группы в которых 4 и более детей, из таблицы видно, что лучше всего кредитные обязательства выполняют заёмщики без детей, скорее всего, это связано с меньшим количеством непредвиденных трат по сравнению с заёмщиками у которых есть дети, однако, на втором месте в рейтенге стоят заёмщики в тремя детьми, возможно эта группа уже опытных и состоявшихся семейных людей, которые научились планировать бюджет с учетом трат и всех непредвиденных расходов на всех детей. Хуже всего кредиты выплачивают группы с одним и двумя детьми, скорее всего, эти группы еще не научились планировать бюджет с учетом трат на детей и непредвиденных расходов или дети появились уже после взятия кредита, таким образом финансовая нагрузка резко возросла и платежи стали слишкои трудными для таких категорий заёмщиков, так же это может быть связанно с уходом в декрет заёмщика или его партнера.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [26]:
data_status = data.groupby('family_status', as_index = False).agg({'debt':['count', 'sum', 'mean']})
data_status.columns = ['Семейное положение','Число заёмщиков','Число должников','Доля должников']
data_status.sort_values(by='Доля должников')

Семейное положение  Число заёмщиков  Число должников  Доля должников
2         вдовец / вдова              951               63        0.066246
1              в разводе             1189               84        0.070648
4        женат / замужем            12261              927        0.075606
3       гражданский брак             4134              385        0.093130
0  Не женат / не замужем             2796              273        0.097639

**Вывод:** наличие серьезных узаконенных отношений на текущий момент или в прошлом дисциплинирует людей и способствует серьезному отношению к их кредитным обязательствам, кроме того, вдовцы и разведенные люди самые лучшие кредитоплательщики потому что привыкли расчитывать на себя и четко знают свои финансовые возможности. Поэтому не женатые и люди живущие в гражданском браке выполняют обязательство перед банком хуже всего.

<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkGreen'>👌 Действительно, оформленные семейные отношения или их прошлый опыт налагают определенную ответственность и приучают к финансовой дисциплине.</font>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [27]:
data_income = data.groupby('total_income_category', as_index = False).agg({'debt':['count', 'sum', 'mean']})
data_income.columns = ['Уровень дохода','Число заёмщиков','Число должников','Доля должников']
data_income.sort_values(by='Доля должников')

Уровень дохода  Число заёмщиков  Число должников  Доля должников
3       до 50 тыс              349               21        0.060172
4     до миллиона             5014              354        0.070602
0  более миллиона               25                2        0.080000
1      до 200 тыс            15921             1353        0.084982
2       до 30 тыс               22                2        0.090909

**Вывод:** Как мы видим, количество заёмщиков в группах, где доход более миллиона  и менее 30 тысяч рублей, не превышает 100 человек, поэтому этих заемщиков мы учитывать при анализе таблицы не будем, так как каждый человек даёт сильный разброс в процентном соотношении. Исходя из данных таблицы можно сделать вывод, что меньше всего просрочек по кредитам у клиентов с уровнем дохода от 30 000 до 50 000 руб в месяц (6.02%), больше всего у клиентов с уровнем дохода от 200 000 до 1 000 000 руб в месяц (8.50%). Эта же категория (от 200 000 до 1 000 000) является наиболее массовой в выборке.

<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkGreen'>👌 Хорошо, что обратили внимание на то, что представленные выборки несбалансированы и какой-то вывод можно делать только по двум категориям с достаточным размером выборок.</font>
<br><font color='DarkMagenta'>👉 В задачах, где требуется анализировать несбалансированные выборки можно, делать дополнительный анализ, разбив данные на равные выборки методом `.qcut`. Подробнее можно прочитать тут: <br>https://dfedorov.spb.ru/pandas/Разделение%20данных%20в%20Pandas%20с%20помощью%20qcut%20и%20cut.html.</font>

<font color='DarkBlue'><b>Комментарий ревьюера 2</b></font><br>
<font color='DarkMagenta'>👉 Можно сделать цикл с перебором количества категорий.</font>

In [29]:
# Комментарий ревьюера 2
temp = data.copy()
start, stop = 3, 8 # начальное и конечное количество категорий
while start <= stop:
    temp['new_category'] = pd.qcut(temp['total_income'], q=start, precision=0)
    que('new_category')
    start += 1

<font color='DarkBlue'><b>Комментарий ревьюера 2</b></font><br>
<font color='DarkMagenta'>👉 Вот, нашлась самая кредито-опасная категория дохода заемщиков, повод для более глубокого анализа.</font>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [28]:
data_purpose = data.groupby('purpose_category', as_index = False).agg({'debt':['count', 'sum', 'mean']})
data_purpose.columns = ['Цель Кредита','Число заёмщиков','Число должников','Доля должников']
data_purpose.sort_values(by='Доля должников')

Цель Кредита  Число заёмщиков  Число должников  Доля должников
1  операции с недвижимостью            10751              780        0.072551
3        проведение свадьбы             2313              183        0.079118
2     получение образования             3988              369        0.092528
0    операции с автомобилем             4279              400        0.093480

**Вывод:** кредиты взятые с целью операций с недвижимостью отдают исправнее всех, хуже всего выплачивают кредиты заёмщики из группы "операции с автомобилем" скорее всего это связано с тем, что покупка недвижимости или ремонт это чаще всего тщательно спланированное и рассчитанное действие 

<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkGreen'>👌 Видимо, кредиты на автомобиль и обучение несут дополнительные риски связанные или с возможной аварией, или с проблемами трудоустройства после обучения.</font>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Так как пропусков у нас было ровно 2174 и только лишь в категориях days_employed и total_income, то можно сделать вывод, что скорее всего данные не учитывались при заполнении или были пропущены. Также можно предположить, что именно по этим категориям произошёл сбой при выгрузке данных.


<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkMagenta'>👉 Существует несколько причин отсутствия данных в датасетах, в том числе:<br>
- Человеческий фактор: Ошибки ввода данных, не желание отвечать на определенные вопросы.
- Технические проблемы: Возможны сбои с оборудованием или программным обеспечением для сбора данных. Отсутствующие данные могут быть удалены в процессе обработки или очистки данных.
- Организационные: Конфиденциальность, могут отсутствовать данные, идентифицирующие отдельных лиц или компании.</font>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Пустые значения можно заменить медианным или средним значением, но Среднеарифметическое значение не подходит при слиьном разбросе данных и всвязи с этим не всегда отображает реальность. Медианное значение отображает более достоверную картину распределения по пользователям, даже если имеется сильный разброс по максимальным и минимальным значениям.

<font color='DarkBlue'><b>Комментарий ревьюера</b></font><br>
<font color='DarkGreen'>👌 Правильно, у средних более высокая чувствительность к выбросам, чем у медиан.<br>Можно попробовать посчитать и сравнить среднюю и медиану для колонки с выбросами, например `days_employed`.</font>

### Шаг 4: общий вывод.

Напишите ваш общий вывод.
В ходе данного исследования была проанализирована выборка текущих клиентов и их задолженностей по платежам.
Для проведения исследования по надежности заёмщиков нам были предоставлены данные с информацией об образовании, семейном положении, уровне образования, цели кредита, общего рабочего стажа, пола, наличия детей, уровня и типа дохода и количестве дней задолженности.

При обработке данных группы с количеством до 100 человек мы не учитывали при анализе, так как в таких группах невозможно вывести достоверную статистику. Также столкнулись с отсутсвием данных в разделе общей прибыли и количестве дней задолженности по кредиту, эти пропуски мы заполнили медианными значениями так как среднеарифметическое значение не подходит при слиьном разбросе данных и всвязи с этим не всегда отображает реальность. Медианное значение отображает более достоверную картину распределения по пользователям, даже если имеется сильный разброс по максимальным и минимальным значениям. Также при предобработке данных мы отсортировали все цели кредита по 4 категориям: Обучение, операции с недвижимостью, автокредитование и свадьба.
Целью анализа данных был поиск оптимального по совокупности факторов клиента на выдачу кредита.

В рамках анализа были проверены следующие гипотезы:

Есть ли зависимость между количеством детей и возвратом кредита в срок?

В рамках данного исследования не удалось оценить группы клиентов с 4 и 5 детьми - выборка оказалась слишком малой для анализа. Категория клиентов без детей оказалась самой многочисленной и процент задержек по кредиту самый низкий - 7.54%. Самый высокий процент задержек по оплатам оказался у клиентов с двумя детьми и составляет 9.45%.
Есть ли зависимость между семейным положением и возвратом кредита в срок?

По результатам проверки данной гипотезы выявлена следующая закономерность - меньше всего просрочек по кредитам у вдов/вдовцев (6.62%), больше всего у клиентов, не состоящих в отношениях (9.76%). Самой массовой категорией являются клиенты в браке, процент просрочек по данной категории составляет 7.56%
Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

По результатам проверки данной гипотезы выявлена следующая закономерность - меньше всего просрочек по кредитам у клиентов с уровнем дохода от 30 000 до 50 000 руб в месяц (6.02%), больше всего у клиентов с уровнем дохода от 200 000 до 1 000 000 руб в месяц (8.50%). Эта же категория (от 200 000 до 1 000 000) является наиболее массовой в выборке.
Категория клиентов с экстремальными значениями доходов (ниже 30 000 руб или выше 1 000 000 руб) являются слишком малой. Достоверно говорить о вероятностях просрочек от этих клиентов нельзя.

Как разные цели кредита влияют на его возврат в срок?

По результатам проверки данной гипотезы выявлена следующая закономерность - меньше всего просрочек по кредитам по займам на недвижимость (7.26%), больше всего по займам на автомобиль (9.35%). Самой массовой категорией являются займы на недвижимость.
Помимо проведенного анализа, возможно стоило бы провести ещё анализ по количеству дней просрочки и объемам просрочек. Этот анализ позволит определить более точно группу которая, с точки зрения задолженности, порождает наибольшие финансовые проблемы.

Итак, самым оптимальным кандидатом на одобрение кредита в банке является клиент без детей, который имел серьезные узаконенные отношения в прошлом или состоит в них сейчай, с уровнем дохода от 30 до 50 тыс. руб., и целью кредитования которого являются любые операции с недвижимостью.
Клиенты, которым нужно выдавать кредиты с особой осторожнсотью и уделить особое внимание при скоринге- это группа людей с детьми, не состоявшие в серьезных узаконенных отношениях, с доходом от 50 до 200 тыс. руб., и чей займ связан с автокредитованием.